src/main/java/rx/internal/operators/OperatorConcat.java

```diff
@@ -179,6 +187,14 @@ void subscribeNext() {
                 }
             }
         }
+        
+        void produced(long c) {
+            arbiter.produced(c);
+            AtomicLong requestedField = requested;
+            if (requestedField.get() != Long.MAX_VALUE) {
+                requestedField.addAndGet(-c);
```

Comment:
This line is correct as it is, what we are talking about is considering the case where one wants to subvert the optimization by racing with get().